R/PipeOpTaskSurvClassifIPCW.R

```diff
@@ -0,0 +1,133 @@
+#' @title PipeOpTaskSurvClassifIPCW
+#' @name mlr_pipeops_trafotask_survclassif_IPCW
+#' @template param_pipelines
+#'
+#' @description
+#' Transform [TaskSurv] to [TaskClassif][mlr3::TaskClassif] using IPCW (Vock et al., 2016).
+#'
+#' @section Dictionary:
+#' This [PipeOp][mlr3pipelines::PipeOp] can be instantiated via the
+#' [dictionary][mlr3misc::Dictionary] [mlr3pipelines::mlr_pipeops]
+#' or with the associated sugar function [mlr3pipelines::po()]:
+#' ```
+#' PipeOpTaskSurvClassifIPCW$new()
+#' mlr_pipeops$get("trafotask_survclassif_IPCW")
+#' po("trafotask_survclassif_IPCW")
+#' ```
+#'
+#' @section Input and Output Channels:
+#' [PipeOpTaskSurvClassifIPCW] has one input channel named "input", and two
+#' output channels, one named "output" and the other "data".
+#'
+#' During training, the "output" is the "input" [TaskSurv] transformed to a
+#' [TaskClassif][mlr3::TaskClassif].
+#' The target column is named `"status"` and indicates whether an event occurred
+#' in each time interval.
+#' The transformed task now has the property "weights".
+#' The "data" is NULL.
+#'
+#' During prediction, the "input" [TaskSurv] is transformed to the "output"
+#' [TaskClassif][mlr3::TaskClassif] with `"status"` as target.
+#' The "data" is a [data.table] containing the "time" of each subject as well
+#' as corresponding "row_ids".
+#' This "data" is only meant to be used with the [PipeOpPredClassifSurvIPCW].
+#'
+#' @section Parameters:
+#' The parameters are
+#'
+#' * `cutoff_time :: numeric()`\cr
+#' Cutoff time for IPCW. Observations with time larger than `cutoff_time` are censored.
+#' * `eps :: numeric()`\cr
+#' Small value to replace `0` survival probabilities with to prevent infinite weights.
+#'
+#' @references
+#' `r format_bib("vock_2016")`
+#'
+#' @family PipeOps
+#' @family Transformation PipeOps
+#' @export
+PipeOpTaskSurvClassifIPCW = R6Class(
+  "PipeOpTaskSurvClassifIPCW",
+  inherit = mlr3pipelines::PipeOp,
+
+  public = list(
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    initialize = function(id = "trafotask_survclassif_IPCW") {
+      param_set = ps(
+        cutoff_time = p_dbl(lower = 0, special_vals = list()),
+        eps = p_dbl(lower = 0, default = 1e-6)
+      )
+      super$initialize(
+        id = id,
+        param_set = param_set,
+        input = data.table(
+          name    = "input",
+          train   = "TaskSurv",
+          predict = "TaskSurv"
+        ),
+        output = data.table(
+          name    = c("output", "data"),
+          train   = c("TaskClassif", "NULL"),
+          predict = c("TaskClassif", "data.table")
+        )
+      )
+    }
+  ),
+
+  private = list(
+    .predict = function(input) {
+      data = input[[1]]$data()
+      data$status = factor(data$status, levels = c("0", "1"))
```

Comment:
is this necessary? is it ensured that input[[1]]$data$status has levels 0 and 1?

Reply:
There was a practical argument to have like this and with this order of levels from what I recall, we had the same issue with the discrete time pipeop